cpu_linux.go

```diff
@@ -0,0 +1,39 @@
+//go:build linux
+// +build linux
+
+//
+// MinIO Object Storage (c) 2021-2022 MinIO, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package madmin
+
+import "github.com/prometheus/procfs/sysfs"
+
+func getCPUFreqGovernor() (string, error) {
+	fs, err := sysfs.NewFS("/sys")
+	if err != nil {
+		return "", err
+	}
+
+	stats, err := fs.SystemCpufreq()
+	if err != nil {
+		return "", err
+	}
+
+	if len(stats) > 0 {
+		return stats[0].Governor, nil // assuming its same on all cpus/cores
```

Comment:
https://www.kernel.org/doc/html/latest/admin-guide/pm/cpufreq.html#generic-scaling-governors

I think for Linux, we can simply check if all cores have 'performance' - if not return any other value

Reply:
Are you suggesting having a `bool` field called `IsGovernorPerformance` instead of a `string` field that returns the configured value?

Reply:
> Are you suggesting having a `bool` field called `IsGovernorPerformance` instead of a `string` field that returns the configured value?

No, my idea is to check for all cores if they are performance or not and returning whatever value which is not performance

Reply:
```
governor := "performance"
for _, st := range stats {
  if st.Governor != "performance" {
    governor = st.Govoernor
  }
}

Reply:
As discussed, have implemented a `bool` based solution. PTAL.